`Cilium-Ginkgo-Tests` hit this:

```
10:18:28      runtime: --- a/Documentation/cmdref/cilium-agent.md
10:18:28      runtime: +++ b/Documentation/cmdref/cilium-agent.md
10:18:28      runtime: @@ -167,7 +167,7 @@ cilium-agent [flags]
10:18:28      runtime:        --tofqdns-enable-dns-compression                Allow the DNS proxy to compress responses to endpoints that are larger than 512 Bytes or the EDNS0 option, if present (default true)
10:18:28      runtime:        --tofqdns-endpoint-max-ip-per-hostname int      Maximum number of IPs to maintain per FQDN name for each endpoint (default 50)
10:18:28      runtime:        --tofqdns-max-deferred-connection-deletes int   Maximum number of IPs to retain for expired DNS lookups with still-active connections (default 10000)
10:18:28      runtime: -      --tofqdns-min-ttl int                           The minimum time, in seconds, to use DNS data for toFQDNs policies (default 3600 )
10:18:28      runtime: +      --tofqdns-min-ttl int                           The minimum time, in seconds, to use DNS data for toFQDNs policies. (default 3600 )
10:18:28      runtime:        --tofqdns-pre-cache string                      DNS cache data at this path is preloaded on agent startup
10:18:28      runtime:        --tofqdns-proxy-port int                        Global port on which the in-agent DNS proxy should listen. Default 0 is a OS-assigned port.
10:18:28      runtime:        --tofqdns-proxy-response-max-delay duration     The maximum time the DNS proxy holds an allowed DNS response before sending it along. Responses are sent as soon as the datapath is updated with the new IP information. (default 100ms)
10:18:31      runtime: HINT: to fix this, run 'make -C Documentation update-cmdref ; git commit Documentation/cmdref --message "docs: Update cmdref"'
10:18:32      runtime: Makefile:30: recipe for target 'check' failed
10:18:32      runtime: make[1]: Leaving directory '/home/vagrant/go/src/github.com/cilium/cilium/Documentation'
10:18:32      runtime: make[1]: *** [check] Error 1
10:18:32      runtime: Makefile:478: recipe for target 'postcheck' failed
10:18:32      runtime: make: *** [postcheck] Error 2
```
https://jenkins.cilium.io/job/Cilium-PR-Ginkgo-Tests-Validated/18980/execution/node/122/log/

(/cc @errordeveloper why wasn't this picked up by the GH helper? Seems like it should have been run 6 days ago if that makes any difference)

--- a/Documentation/cmdref/cilium-agent.md
+++ b/Documentation/cmdref/cilium-agent.md
@@ -165,11 +165,9 @@ cilium-agent [flags]
       --state-dir string                              Directory path to store runtime state (default "/var/run/cilium")
       --tofqdns-dns-reject-response-code string       DNS response code for rejecting DNS requests, available options are '[nameError refused]' (default "refused")
       --tofqdns-enable-dns-compression                Allow the DNS proxy to compress responses to endpoints that are larger than 512 Bytes or the EDNS0 option, if present (default true)
-      --tofqdns-enable-poller                         Enable proactive polling of DNS names in toFQDNs.matchName rules.
-      --tofqdns-enable-poller-events                  Emit DNS responses seen by the DNS poller as Monitor events, if the poller is enabled. (default true)
       --tofqdns-endpoint-max-ip-per-hostname int      Maximum number of IPs to maintain per FQDN name for each endpoint (default 50)
       --tofqdns-max-deferred-connection-deletes int   Maximum number of IPs to retain for expired DNS lookups with still-active connections (default 10000)
-      --tofqdns-min-ttl int                           The minimum time, in seconds, to use DNS data for toFQDNs policies. (default 600 when --tofqdns-enable-poller, 3600 otherwise)
+      --tofqdns-min-ttl int                           The minimum time, in seconds, to use DNS data for toFQDNs policies. (default 3600 )
       --tofqdns-pre-cache string                      DNS cache data at this path is preloaded on agent startup
       --tofqdns-proxy-port int                        Global port on which the in-agent DNS proxy should listen. Default 0 is a OS-assigned port.
       --tofqdns-proxy-response-max-delay duration     The maximum time the DNS proxy holds an allowed DNS response before sending it along. Responses are sent as soon as the datapath is updated with the new IP information. (default 100ms)

--- a/Documentation/install/upgrade.rst
+++ b/Documentation/install/upgrade.rst
@@ -444,6 +444,10 @@ Deprecated options
   1.6. The ``access-log`` option to log to a file has been removed.
 * ``--disable-k8s-services`` option from cilium-agent has been deprecated
   and will be removed in Cilium 1.9.
+* ``--tofqdns-enable-poller``: This option has been deprecated and will be
+  removed in Cilium 1.9
+* ``--tofqdns-enable-poller-events``: This option has been deprecated and will
+  be removed in Cilium 1.9
 
 Renamed Metrics
 ~~~~~~~~~~~~~~~

--- a/Documentation/policy/language.rst
+++ b/Documentation/policy/language.rst
@@ -479,8 +479,8 @@ provided in DNS responses are allowed by Cilium in a similar manner to IPs in
 or are not know a priori, or when DNS is more convenient. To enforce policy on
 DNS requests themselves, see `Layer 7 Examples`_.
 
-IP information is captured from DNS responses per-Endpoint via a `DNS Proxy`_
-or `DNS Polling`_. An L3 `CIDR based`_ rule is generated for every ``toFQDNs``
+IP information is captured from DNS responses per-Endpoint via a `DNS Proxy`_.
+An L3 `CIDR based`_ rule is generated for every ``toFQDNs``
 rule and applies to the same endpoints. The IP information is selected for
 insertion by ``matchName`` or ``matchPattern`` rules, and is collected from all
 DNS responses seen by Cilium on the node. Multiple selectors may be included in
@@ -517,8 +517,6 @@ IPs to be allowed are selected via:
   * ``*`` alone matches all names, and inserts all cached DNS IPs into this
     rule.
 
-.. note:: `DNS Polling`_ will not poll ``matchPattern`` entries even if they
-          are literal DNS names.
 
 Example
 ~~~~~~~
@@ -554,16 +552,13 @@ state will be have ``source: connection`` with a single IP listed within the
 A minimum TTL is used to ensure a lower time bound to DNS data expiration, and
 IPs allowed by a ``toFQDNs`` rule will be allowed at least this long It can be
 configured with the ``--tofqdns-min-ttl`` CLI option. The value is in integer
-seconds and must be 1 or more. The default is 1 hour, or 10 minutes when `DNS
-Polling`_ is enabled.
+seconds and must be 1 or more, the default is 1 hour.
 
 Some care needs to be taken when setting ``--tofqdns-min-ttl`` with DNS data
 that returns many distinct IPs over time. A long TTL will keep each IP cached
 long after the related connections have terminated. Large numbers of IPs each
 have corresponding Security Identities and too many may slow down Cilium policy
-regeneration. This can be especially pronounced when using `DNS Polling`_ to
-obtain DNS data. In such cases a shorter minimum TTL is recommended, as `DNS
-Polling`_ will recover up-do-date IPs regularly.
+regeneration.
 
 Managing Short-Lived Connections & Maximum IPs per FQDN/endpoint
 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
@@ -1007,8 +1002,8 @@ agent respecting TTL.
 
 .. _DNS Proxy:
 
-DNS Proxy (preferred)
-"""""""""""""""""""""
+DNS Proxy 
+"""""""""
   A DNS Proxy intercepts egress DNS traffic and records IPs seen in the
   responses. This interception is, itself, a separate policy rule governing the
   DNS requests, and must be specified separately. For details on how to enforce
@@ -1040,54 +1035,6 @@ DNS Proxy (preferred)
 .. only:: epub or latex
 
         .. literalinclude:: ../../examples/policies/l7/dns/dns-visibility.json
-
-.. _DNS Polling:
-
-DNS Polling
-"""""""""""
-  DNS Polling periodically issues a DNS lookup for each ``matchName`` from
-  cilium-agent. The result is used to regenerate endpoint policy.  Despite the
-  name, the ``matchName`` field does not have to be a fully-qualified domain
-  name. In cases where search domains are configured for cilium-agent, the DNS
-  lookups from Cilium will not be qualified and will utilize the search list.
-  Unqualified names must be matched as-is by ``matchPattern`` in order to
-  insert related IPs.
-
-  DNS lookups are repeated with an interval of 5 seconds, and are made for
-  A(IPv4) and AAAA(IPv6) addresses. Should a lookup fail, the most recent IP
-  data is used instead. An IP change will trigger a regeneration of the Cilium
-  policy for each endpoint and increment the per cilium-agent policy repository
-  revision.
-
-  Polling may be enabled by the ``--tofqdns-enable-poller`` cilium-agent
-  CLI option. It is disabled by default.
-
-  The DNS polling implementation is very limited. It may not behave as expected.
-
-  #. The DNS polling is done from the cilium-agent process. This may result in
-     different IPs being returned in the DNS response than those seen by an
-     application.
-
-  #. When using DNS Polling with DNS responses that return a new IP on every
-     query, the IP being whitelisted may differ from the one used for
-     connections by applications. This is because the application will make
-     a DNS query independent from the poll.
-
-  #. When DNS lookups return many distinct IPs over time, large values of
-     ``--tofqdns-min-ttl`` may result in unacceptably slow policy
-     regeneration. See `DNS and Long-Lived Connections`_ for details.
-
-  #. The lookups from Cilium follow the configuration of the environment it
-     is in via ``/etc/resolv.conf``. When running as a kubernetes pod, the
-     contents of ``resolv.conf`` are controlled via the ``dnsPolicy`` field of a
-     spec. When running directly on a host, it will use the host's file.
-     Irrespective of how the DNS lookups are configured, TTLs and caches on the
-     resolver will impact the IPs seen by the cilium-agent lookups.
-
-.. note:: Connections to the DNS resolver must be explicitly whitelisted to
-          allow DNS queries. This is independent of the source of DNS
-          information, whether from polling or the DNS proxy.
-
 
 Alpine/musl deployments and DNS Refused
 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

--- a/daemon/cmd/daemon_main.go
+++ b/daemon/cmd/daemon_main.go
@@ -674,7 +674,7 @@ func init() {
 	flags.MarkHidden(option.CMDRef)
 	option.BindEnv(option.CMDRef)
 
-	flags.Int(option.ToFQDNsMinTTL, 0, fmt.Sprintf("The minimum time, in seconds, to use DNS data for toFQDNs policies. (default %d when --tofqdns-enable-poller, %d otherwise)", defaults.ToFQDNsMinTTLPoller, defaults.ToFQDNsMinTTL))
+	flags.Int(option.ToFQDNsMinTTL, 0, fmt.Sprintf("The minimum time, in seconds, to use DNS data for toFQDNs policies. (default %d )", defaults.ToFQDNsMinTTL))
 	option.BindEnv(option.ToFQDNsMinTTL)
 
 	flags.Int(option.ToFQDNsProxyPort, 0, "Global port on which the in-agent DNS proxy should listen. Default 0 is a OS-assigned port.")
@@ -682,9 +682,11 @@ func init() {
 
 	flags.Bool(option.ToFQDNsEnablePoller, false, "Enable proactive polling of DNS names in toFQDNs.matchName rules.")
 	option.BindEnv(option.ToFQDNsEnablePoller)
+	flags.MarkDeprecated(option.ToFQDNsEnablePoller, "This option has been deprecated and will be removed in v1.9")
 
 	flags.Bool(option.ToFQDNsEnablePollerEvents, true, "Emit DNS responses seen by the DNS poller as Monitor events, if the poller is enabled.")
 	option.BindEnv(option.ToFQDNsEnablePollerEvents)
+	flags.MarkDeprecated(option.ToFQDNsEnablePollerEvents, "This option has been deprecated and will be removed in v1.9")
 
 	flags.StringVar(&option.Config.FQDNRejectResponse, option.FQDNRejectResponseCode, option.FQDNProxyDenyWithRefused, fmt.Sprintf("DNS response code for rejecting DNS requests, available options are '%v'", option.FQDNRejectOptions))
 	option.BindEnv(option.FQDNRejectResponseCode)

